src/QsCompiler/DataStructures/SyntaxTree.fs

```diff
@@ -720,6 +753,34 @@ type SpecializationImplementation =
     /// indicates that the specialization is to be generated by the compiler according to the given functor generator directive
     | Generated of QsGeneratorDirective // Invert and Distribute will be replaced by Provided before sending to code gen
 
+/// <summary>
+/// The schema for <see cref="QsSpecialization"/> that is used with JSON serialization.
+/// </summary>
+[<CLIMutable>]
+[<DataContract>]
+type internal QsSpecializationSchema =
```

Comment:
I think it would make more sense to have these in the Serialization file - are they needed here?